lib/openhab/log/logger.rb

```diff
@@ -148,38 +152,26 @@ class << self
       #
       # @return [Logger] for the supplied name
       #
-      def logger(klass)
+      def logger_for_class(klass)
         if klass.respond_to?(:java_class) &&
            klass.java_class &&
            !klass.java_class.name.start_with?('org.jruby.Ruby')
           klass = klass.java_class
         end
-        name = klass.name
+        name = klass.name if klass.name != 'Object'
         @loggers[name] ||= Log.logger_for(name)
       end
 
       #
-      # Configure a logger for the supplied class name
-      #
-      # @param [String] name to configure logger for
-      #
-      # @return [Logger] for the supplied classname
-      #
-      def logger_for(name)
-        configure_logger_for(name)
-      end
-
-      private
-
-      #
-      # Configure a logger for the supplied classname
+      # Configure a logger for the supplied name
       #
       # @param [String] classname to create logger for
       #
-      # @return [Logger] Logger for the supplied classname
+      # @return [Logger] Logger for the supplied name
       #
-      def configure_logger_for(name)
+      def logger_for(name)
         log_prefix = Configuration.log_prefix
+        Logger.new('XXX').error(caller_locations.to_s) if name == 'Object'
```

Comment:
If name is Object, object should be removed from the logger name.

Reply:
Oops this was just for a quick debugging, not supposed to be included.